guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java

```diff
@@ -126,10 +139,28 @@ public void sendHeaders(final Metadata responseHeaders) {
                 try {
                     grpcFilters.forEach(filter -> filter.doProcessResponseHeaders(responseHeaders));
                     super.sendHeaders(responseHeaders);
+                } catch (Exception e) {
+                    handleException(call, grpcFilters, e);
                 } finally {
                     detachContext(contextWithHeaders, previous);    // detach headers from gRPC context
                 }
             }
+
+//            @Override
+//            public void close(Status status, Metadata trailers) {
+//                Context previous = attachContext(contextWithHeaders);   // attaching headers to gRPC context
+//                error("@@@@@@@@@@@@@@@@@@ got inside close");
+//                try {
+//                    if (!status.isOk()){
+//                        grpcFilters.forEach(filter -> filter.doHandleException(new StatusRuntimeException(status,trailers)));
+//                    }
+//                    super.close(status, trailers);
+//                } catch (Exception e) {
+//                    handleException(call, grpcFilters, e);
+//                } finally {
+//                    detachContext(contextWithHeaders, previous);    // detach headers from gRPC context
+//                }
+//            }
```

Comment:
with close() we can get the exact Status and print it in the access logs.
Need to check how to get this with onCancel() because close() is not guaranteed to be called always in the javadocs.